Antidote/AddFriendViewController.m

```diff
@@ -61,8 +68,45 @@ - (void)viewDidLayoutSubviews
     [self adjustSubviews];
 }
 
+- (void)dealloc
+{
+    [self unsubscribeNotifications];
+}
+
+#pragma mark -  Notifications
+
+- (void)subscribeNotifications
+{
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(keyboardWillShow)
+                                                 name:UIKeyboardWillShowNotification
+                                               object:nil];
+    
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(keyboardWillHide)
+                                                 name:UIKeyboardWillHideNotification
+                                               object:nil];
+}
+
+- (void)unsubscribeNotifications
+{
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
 #pragma mark -  Actions
 
+- (void)keyboardWillShow
+{
+    self.scrollView.scrollEnabled = NO;
+}
+
+- (void)keyboardWillHide
+{
+    self.scrollView.scrollEnabled = YES;
+    
+    [self finishEditingButtonPressed];
```

Comment:
The button wasn't actually pressed. As this method only resigns responders, it would be better to rename it to something like `resignTextViewResponders`